packages/emitter-framework/src/typescript/components/value-expression.tsx

```diff
@@ -0,0 +1,54 @@
+import { Children } from "@alloy-js/core";
+import * as ts from "@alloy-js/typescript";
+import { Value } from "@typespec/compiler";
+
+/**
+ * Properties for the {@link ValueExpression} component.
+ */
+interface ValueExpressionProps {
+  /**
+   * The TypeSpec value to be converted to a JavaScript expression.
+   */
+  value: Value;
+}
+
+/**
+ * Generates a JavaScript value expression from a TypeSpec value.
+ * @param props properties for the value expression
+ * @returns {@link Children} representing the JavaScript value expression
+ */
+export function ValueExpression(props: ValueExpressionProps): Children {
+  switch (props.value.valueKind) {
+    case "StringValue":
+    case "BooleanValue":
+    case "NullValue":
+      return <ts.ValueExpression jsValue={props.value.value} />;
+    case "NumericValue":
+      if (props.value.value.asNumber()) {
+        return <ts.ValueExpression jsValue={props.value.value.asNumber()} />;
+      }
+      return <ts.ValueExpression jsValue={props.value.value.asBigInt()} />;
+    case "ArrayValue":
+      return (
+        <ts.ArrayExpression
+          jsValue={props.value.values.map((v) => (
+            <ValueExpression value={v} />
+          ))}
+        />
+      );
+    case "ScalarValue":
+      if (props.value.value.name === "fromISO") {
+        return <ValueExpression value={props.value.value.args[0]} />;
+      } else {
+        throw new Error("Unsupported scalar constructor: " + props.value.value.name);
```

Comment:
```suggestion
        compilerAssert(false, `Unsupported scalar constructor: ${props.value.value.name}`, props.value)
```

this way when it crash it will include the location of the value

Reply:
the `compilerAssert` function seems to be concerned with `Use this to report bugs in the compiler, and not errors in the source code being compiled.` - would this be a compiler bug if we got here?

Reply:
yeah the name maybe is not the best and we could rexport a more generic function but its been used  for those purposes.

In general in typespec you should never throw errors unless you know it wont happen unless there is a bug and seems like we have a bit of an issue we'd need to solve for emitter framework error handling here. 

But for now this at least will give a better error to the user so they can understand what to change